scripts/generate_example_pngs.R

```diff
@@ -0,0 +1,191 @@
+  )
+
+# Revise the list of topic names by removing the excluded ones
+fn_names <- fn_names %>% base::setdiff(name_exclusions)
+
+# Create a `save_with_webshot()` function that allows us to
+# resize the images before writing to disk
+save_with_webshot <- function(data,
+                              filename,
+                              path = NULL,
+                              ...,
+                              zoom = 2,
+                              expand = 5,
+                              resize = NULL) {
+
+  gtsave_filename <- function(path, filename) {
+
+    if (!is.null(path)) {
+      filename <- file.path(path, filename)
+    }
+
+    filename %>% gt:::path_expand()
+  }
+
+  filename <- gtsave_filename(path = path, filename = filename)
+
+  # Create a temporary file with the `html` extension
+  tempfile_ <- tempfile(fileext = ".html")
+
+  # Reverse slashes on Windows filesystems
+  tempfile_ <-
+    tempfile_ %>%
+    gt:::tidy_gsub("\\\\", "/")
+
+  # Save gt table as HTML using the `gt_save_html()` function
+  data %>% gt:::gt_save_html(filename = tempfile_, path = NULL)
+
+  # Saving an image requires the webshot package; if it's
+  # not present, stop with a message
+  if (!requireNamespace("webshot", quietly = TRUE)) {
+
+    stop("The `webshot` package is required for saving images of gt tables.",
+         call. = FALSE)
+
+  } else {
+
+    if (is.null(resize)) {
+      webshot::webshot(
+        url = paste0("file:///", tempfile_),
+        file = filename,
+        zoom = zoom,
+        expand = expand,
+        ...
+      )
+    } else {
+      webshot::webshot(
+        url = paste0("file:///", tempfile_),
+        file = filename,
+        zoom = zoom,
+        expand = expand,
+        ...
+      ) %>%
+        webshot::resize(resize)
+    }
+  }
+}
+
+# Declare a fixed image width
+image_width <- 1100
+
+# Obtain a temporary directory name
+temp_dir <- tempdir()
+
+# Generate images in `temp_dir`
+for (fn_name in fn_names) {
+
+  # Get the initial list of objects in the environment
+  obj_list_initial <- ls()
+
+  # Evaluate the examples in the examples for each topic/function; the
+  # objects we'd want to create images from all begin with `tab_`
+  utils::example(topic = fn_name, package = "gt", character.only = TRUE, give.lines = FALSE)
+
+  # Get a vector of new objects created
+  new_obj <- base::setdiff(ls(), obj_list_initial)
+
+  # Get a separate vector of objects that begin with `tab_`
+  tab_obj <- new_obj %>% grep("^tab_.\\d*?", ., value = TRUE)
+
+  # For each of the `tab_` objects, create an image using webshot
+  for (i in seq_along(tab_obj)) {
+
+    # Create a systematic file name so that the help file
+    # can access multiple images in the correct order
+    filename <- file.path(temp_dir, paste0("man_", fn_name, "_", i, ".png"))
+
+    # Save the image with a high `zoom` value for clarity,
+    # but resized down to 50% of the original
+    save_with_webshot(
+      data = get(tab_obj[i]),
```

Comment:
Maybe validate that `tab_obj[i]` is a gt object?

Reply:
Now done before using **webshot**.